### PR TITLE
Add txredisapi to matrix-synapse dependencies

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Motivation for this change
When using workers with synapse txredisapi is required.

###### Things done

I've been using basically this for half a year+ with a module with worker support.

I'm not sure if it makes sense at all to do the unit testing like this for the txredisapi package. If someone has an opinion on keeping/improving/or removing I'm all ears.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128048` at 44a982a0 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.txredisapi</li>
    <li>python39Packages.txredisapi</li>
  </ul>
</details>

**Review by fadenb — APPROVED** [review]

**Review by Ma27 — APPROVED** [review]

File: pkgs/development/python-modules/txredisapi/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "IlyaSkriblovsky";
+    repo = "txredisapi";
+    rev = "1.4.7";
+    sha256 = "1f7j3c5l7jcfphvsk7nqmgyb4jaydbzq081m555kw0f9xxak0pgq";
+  };
+
+  propagatedBuildInputs = [ six twisted ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/txredisapi/default.nix:16]: Why are the tests disabled?

File: pkgs/development/python-modules/txredisapi/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  pythonImportsCheck = [ "txredisapi" ];
+
+  passthru.tests.unit-tests = nixosTests.txredisapi;
+
+  meta = with lib; {
+    homepage = "https://github.com/IlyaSkriblovsky/txredisapi";
+    description = "non-blocking redis client for python";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/txredisapi/default.nix:23]: description should start with a capital letter.

File: nixos/tests/txredisapi.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  nodes = {
+    machine =
+      { pkgs, ... }:
+
+      {
+        services.redis.enable = true;
+        services.redis.unixSocket = "/run/redis/redis.sock";
+
+        environment.systemPackages = with pkgs; [ (python38.withPackages (ps: [ ps.twisted ps.txredisapi ps.mock ]))];
```
Comment (SuperSandro2000) [inline nixos/tests/txredisapi.nix:16]: that should probably just be python3

File: pkgs/development/python-modules/txredisapi/default.nix (hunk 1 above)
Comment (dali99) [reply pkgs/development/python-modules/txredisapi/default.nix:16]: Because there are no tests. The only tests are the unit tests ran in the nixos test

Comment (SuperSandro2000) [reply pkgs/development/python-modules/txredisapi/default.nix:16]: I didn't want you to tell me that but rather add a comment with that information.

Comment (dali99) [reply pkgs/development/python-modules/txredisapi/default.nix:16]: Oh, okay. In the future maybe you could say something like "You should add a comment explaining why the tests are disabled", thank you though

**Pull request merged**